### PR TITLE
Undefined variable warnings: fixes

### PR DESCRIPTION
**Pull request by Terramorpha**
Currently, the new undefined variable warnings are not used during the compilation of gambit itself.

### CODE REVIEW

File: lib/module-common.mk.in [hunk 1]
```diff
@@ -224,7 +224,7 @@ modules-pre:
 	    fi; \
 	    if test "$$need_to_build_module" = "yes"; then \
 	      echo ">>>> building module $$modref for $$target"; \
-	      @SETDLPATH@ $(rootfromhere)/gsc/gsc@exe@ -:~~bin=$(srcdirpfx)$(rootfromhere)/bin,~~lib=$(srcdirpfx)$(rootfromhere)/lib,~~include=$(srcdirpfx)$(rootfromhere)/include -f -e "(let ((x (##build-module \"$$mod_dir$$main_mod\" (quote $$target) (quote ((module-ref $$modref)))))) (if (not x) (begin (println \"*** module failed to build\") (exit 1))))"; \
+	      @SETDLPATH@ $(rootfromhere)/gsc/gsc@exe@ -:~~bin=$(srcdirpfx)$(rootfromhere)/bin,~~lib=$(srcdirpfx)$(rootfromhere)/lib,~~include=$(srcdirpfx)$(rootfromhere)/include -f -e "(let ((x (##build-module \"$$mod_dir$$main_mod\" (quote $$target) (quote ((module-ref $$modref) (warnings)))))) (if (not x) (begin (println \"*** module failed to build\") (exit 1))))"; \
```
Comment (feeley) [inline lib/module-common.mk.in:227]: Adding `(warnings)` is premature. Only when the building of the modules generates no warnings should we officially add this to the repo.

**Review by feeley — CHANGES_REQUESTED** [review]

**Comment by gambiteer** [comment]
This is nice.  I still see some repeated warnings, which I don't understand:
```
heine:~/programs/gambit/gambit> sort crapolita2 | uniq -c | sort -n | grep WARNING | tail -10
      1 ****** WARNING IN "/home/lucier/programs/gambit/gambit/gsc/../gsc/_envadt.scm"@16.54 -- "c#next-var-stamp" is not defined
      2 *** WARNING IN "/home/lucier/programs/gambit/gambit/gsc/../gsc/_ptreeadt.scm"@128.54 -- "c#next-node-stamp" is not defined
      2 *** WARNING IN "/home/lucier/programs/gambit/gambit/gsc/../gsc/_ptreeadt.scm"@160.54 -- "c#next-node-stamp" is not defined
      2 *** WARNING IN "/home/lucier/programs/gambit/gambit/gsc/../gsc/_ptreeadt.scm"@62.54 -- "c#next-node-stamp" is not defined
      3 *** WARNING IN "/home/lucier/programs/gambit/gambit/gsc/../gsc/_ptreeadt.scm"@110.55 -- "c#next-node-stamp" is not defined
      5 *** WARNING IN "/home/lucier/programs/gambit/gambit/gsc/../gsc/_gvmadt.scm"@32.32 -- "c#enter-opnd" is not defined
      5 *** WARNING IN "/home/lucier/programs/gambit/gambit/gsc/../gsc/_gvmadt.scm"@34.39 -- "c#*opnd-table*" is not defined
      6 *** WARNING IN "/home/lucier/programs/gambit/gambit/gsc/../gsc/_envadt.scm"@16.54 -- "c#next-var-stamp" is not defined
      7 *** WARNING IN "/home/lucier/programs/gambit/gambit/gsc/../gsc/_gvmadt.scm"@37.38 -- "c#enter-opnd" is not defined
      8 *** WARNING IN "/home/lucier/programs/gambit/gambit/gsc/../gsc/_gvmadt.scm"@40.40 -- "c#*opnd-table*" is not defined
```
The build commands were
```
make clean\
&& rm -rf boot gsc-boot\
&& git pull\
&& make -j8 core\
&& make -j8 modules\
&& make doc\
&& make check
```

**Comment by feeley** [comment]
You get multiple errors because there were multiple modules in `gsc/` that were compiled and contain these undefined references. This will help create appropriate `(namespace ("c#" ...))` declarations to avoid these warnings.

**Comment by gambiteer** [comment]
Was _gvmadt.scm compiled multiple times? Because there are 8 warnings that c#*opnd-table* is not defined in that file.

**Comment by feeley** [comment]
8 "includes".

**Pull request merged**